pkg/broker/handler/pool/fanout/pool.go

```diff
@@ -54,6 +54,25 @@ type SyncPool struct {
 	deliverClient ceclient.Client
 }
 
+type handlerCache struct {
+	handler.Handler
+	b *config.Broker
+}
+
+func (hc *handlerCache) requiresRestart(b *config.Broker) bool {
+	if !hc.IsAlive() {
+		return true
+	}
+	if b == nil || b.DecoupleQueue == nil {
```

Comment:
I suggest moving this to a `shouldDelete` function because in this case we don't want to `restart` but simply `delete` the handler.

Reply:
We never really reuse a handler. Restart here means delete and recreate.

Reply:
It seems the code will fail if b.DecoupeQueue is nil when we try to use it to create the handler.

Reply:
Let me rename it to `shouldReconcile`.

`b.DecoupleQueue.Topic` empty shouldn't really happen in this case. If it really happens, then let the handler creation fail. If the data eventually gets correct, at that point a handler will be created.

Reply:
Also added a comment.

Reply:
It doesn't look we will return an error gracefully if `b.DecoupleQueue == nil`, see line #149 in RangeBroker. This probably doesn't happen in practice but also it's hard to ensure the behavior of the controller won't change. 

It doesn't need to be solved in this PR but maybe we should have a `Validate` method on the config and the controller only persists validated results.

Reply:
Created an issue: https://github.com/google/knative-gcp/issues/939